dli_app/templates/reports/all.html

```diff
@@ -20,21 +24,38 @@
           <td>{{ report.id }}</td>
           <td>{{ report.name }}</td>
           <td>{{ report.tagnames|join(', ') }}</td>
-          <td><a href="{{ url_for('reports.view_report', report_id=report.id) }}" class="btn btn-info btn-xs">View</a></td>
           <td>
-          {% if report not in current_user.favorite_reports %}
-            <form method="POST" action="{{ url_for('reports.favorite_report', report_id=report.id) }}">
-              <button type="submit" class="btn btn-success btn-xs">
-                <span class="glyphicon glyphicon-star-empty" aria-hidden="true"></span>
-                  Favorite
-          {% else %}
-            <form method="POST" action="{{ url_for('reports.unfavorite_report', report_id=report.id) }}">
-              <button type="submit" class="btn btn-warning btn-xs">
-                <span class="glyphicon glyphicon-star" aria-hidden="true"></span>
-                  Unfavorite
-          {% endif %}
-              </button>
-            </form>
+            {% if report not in current_user.favorite_reports %}
+              <form method="POST" action="{{ url_for('reports.favorite_report', report_id=report.id) }}">
+                <button type="submit" class="btn btn-success btn-xs">
+                  <span class="glyphicon glyphicon-star-empty" aria-hidden="true"></span> Favorite
+                </button>
+              </form>
+
+              <form method="POST" action="{{ url_for('reports.delete_report', report_id=report.id) }}">
+                <button type="submit" class="btn btn-danger btn-xs">Delete</button>
+              </form>
+
+              <div class="btn-group">
+                <a href="{{ url_for('reports.view_report', report_id=report.id) }}" class="btn btn-info btn-xs">View</a>
+                <a href="{{ url_for('reports.submit_report_data', dept_id=current_user.dept_id) }}" class="btn btn-default btn-xs">Submit Data</a>
```

Comment:
dept_id parameter is not needed.
At least in this sprint, we don't really have a way of submitting data on a per-report basis. It only works on a site-wide scale. You could actually put the "Submit report data" button at the top of the page if you want.